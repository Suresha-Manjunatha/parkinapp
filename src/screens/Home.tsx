/** @format */

import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import React, { FC, useEffect, useState } from "react";
import { Button, SafeAreaView, TextInput } from "react-native";
import { style } from "../styles/style";

type routesType = {
  Lots: {
    lots: number;
  };
};

const Home: FC = () => {
  const [lots, setLots] = useState<number>(0);

  const navigation = useNavigation<NativeStackNavigationProp<routesType>>();

  return (
    <SafeAreaView testID="home" style={style.container}>
      <TextInput
        testID="txt"
        placeholder="Enter number of Parking Lots"
        placeholderTextColor={"grey"}
        keyboardType="numeric"
        onChangeText={(text) => setLots(Number(text))}
        style={style.input}
      />
      <Button
        testID="btn"
        disabled={lots == 0}
        title="Submit"
        onPress={() => {
          navigation.navigate("Lots", { lots: lots });
        }}
        color="orange"
      />
    </SafeAreaView>
  );
};

export default Home;
