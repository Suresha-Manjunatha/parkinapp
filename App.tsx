import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import Home from "./src/screens/Home";
import Lots from "./src/screens/Lots";

export default function App() {
  const Stack = createNativeStackNavigator();
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={Home} />
        <Stack.Screen name="Lots" component={Lots} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
