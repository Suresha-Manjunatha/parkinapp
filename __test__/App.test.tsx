import { render } from "@testing-library/react-native";
import App from "../App";
import Home from "../src/screens/Home";

test("test app", () => {
  render(<App />);
});

test("test app", () => {
  const tree = render(<App />);
  expect(tree).toMatchSnapshot();
});
