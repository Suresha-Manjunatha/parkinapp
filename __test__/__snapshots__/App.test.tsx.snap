// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "lightskyblue",
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            }
          }
          testID="home"
        >
          <TextInput
            keyboardType="numeric"
            onChangeText={[Function]}
            placeholder="Enter number of Parking Lots"
            placeholderTextColor="grey"
            style={
              Object {
                "borderColor": "grey",
                "borderRadius": 5,
                "borderWidth": 1,
                "marginBottom": 10,
                "padding": 5,
              }
            }
            testID="txt"
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="btn"
          >
            <View
              style={
                Array [
                  Object {},
                  Object {},
                ]
              }
            >
              <Text
                disabled={true}
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "orange",
                    },
                    Object {
                      "color": "#cdcdcd",
                    },
                  ]
                }
              >
                Submit
              </Text>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
